### ISSUE TITLE
Useless semicolons

### ISSUE THREAD

**Issue by leonardo-m** [rust-lang/rust-clippy#3305]
I suggest to give some kind of warning or hint to remove the double ;; here:

```
fn main() {
    println!("hello");;
}
```

The D language is different from Rust so the situation isn't directly comparable, but in a case like this gives a warning like this:

temp.d(2,22): Warning: use '{ }' for an empty statement, not a ';'

**Comment by estebank** [comment]
Adding a different case from https://github.com/rust-lang/rust/issues/56338:

```
enum Foo { A, B }

fn bar(f: Foo) {
    match f {
        Foo::A => {},
        Foo::B => {},
    };
}

fn main() {
    bar(Foo::A);
    bar(Foo::B);
}
```

**Comment by leonardo-m** [comment]
In my opinion this is fitter in rustc more than in Clippy. But better to have it in Clippy than having it nowhere.

**Comment by JohnTitor** [comment]
The first case is caught by rustc lint (`redundant_semicolon`) and Clippy lint (`clippy::no_effect`).

**Comment by phansch** [comment]
Shouldn't this be just an extension of the `redundant_semicolons` lint then? It would feel weird to have `clippy::redundant_semicolons` and `redundant_semicolons` in rustc.

**Comment by camsteffen** [comment]
This applies to anything with a block: `if`, `if let`, `while`, `for`. I think it should be a separate lint `needless_semicolon_after_block`.

**Comment by dswij** [comment]
There is now a rustc lint `redundant_semicolons` for this
